shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/generic/DataTypeLengthSegment.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sql.parser.sql.segment.generic;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.shardingsphere.sql.parser.sql.segment.SQLSegment;
+
+import java.util.LinkedList;
+
+@Getter
+@Setter
+public final class DataTypeLengthSegment implements SQLSegment {
+    
+    private int startIndex;
+    
+    private int stopIndex;
+    
+    private int length;
+    
+    private LinkedList<Integer> lengthPair;
```

Comment:
Consider the samples of length may be (1) or (2, 1), I suggest the members of `DataTypeLengthSegment ` to be,
```
 private int startIndex;

     private int stopIndex;

     private int firstNumber;
 
     private int secondNumber;

     public Optional<int> getSecondNumber(){}
```

Reply:
I think it is better as a whole. When using the whole length, users can use it according to their needs.

Reply:
What I concerned about is that `length` and `lengthPair`will confused user who considers `length` and `lengthPair` can have value at the same time. But the fact is that either `length` or `lengthPairan' can have the value at one time. If the user calls `getLength()` and `getLengthPair()` together, what will happen?

Reply:
If just have one num,lengthPair is null, if have two, length is null.